### DISCUSSION TITLE
Add CPU model and OS name values to dBus properties.

### DISCUSSION THREAD

**Discussion by smallorange** [fwupd/fwupd#5262] [Ideas]
OS name and CPU model are required when the user exports the detailed security attribute information through gnome-control-panel. Currently, the glibtop and g_get_os_info() are used to get the data. Since the device security panel was used to show the fwupd information, if that information could be got from fwupd, the presentation of the data could be managed by fwupd.

I've implemented the function shown bellow to get the CPU model and OS name.

- Get OS Name
```
static gchar *
get_os_name (void)
{
  g_autofree gchar *name = NULL;
  g_autofree gchar *version_id = NULL;
  g_autofree gchar *pretty_name = NULL;

  name = g_get_os_info (G_OS_INFO_KEY_NAME);
  version_id = g_get_os_info (G_OS_INFO_KEY_VERSION_ID);
  pretty_name = g_get_os_info (G_OS_INFO_KEY_PRETTY_NAME);

  if (pretty_name)
    return g_steal_pointer (&pretty_name);
  else if (name && version_id)
    return g_strdup_printf ("%s %s", name, version_id);
  else
    return g_strdup (_("Unknown"));
}
```
- Get CPU model
```
static gchar*
cpu_get_model ()
{
  gchar *model;
  const glibtop_sysinfo * sysinfo;

  glibtop_init();
  sysinfo = glibtop_get_sysinfo ();
  model = g_strdup (g_hash_table_lookup (sysinfo->cpuinfo [1].values, "model name"));
  glibtop_close ();

  return model;
}
```
- The output strings for the current implementation are:
```
Processor:                                                      Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz
OS:                                                             Fedora Linux 38 (Workstation Edition Prerelease)
```

Could you add CPU model and OS name to the dBus properties?

Thank you :)

**Comment by hughsie** [comment]
@superm1 how do you feel about a `HostDistro` property on the root interface with something like `Fedora Linux 38 (Workstation Edition Prerelease)` and a `HostCpu` with a property `Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz` -- the latter probably populated from the CPU plugin I guess.

**Reply by superm1** [reply]
No concerns

**Reply by superm1** [reply]
Maybe `HostOS` though rather than distro.
